### PR TITLE
feat: added support Azure DevOps git provider in factory tests

### LINKED ISSUE #22348: Support Azure DevOps git provider in factory E2E typescript tests
### Is your task related to a problem? Please describe

It's needed to automate testing of factory with Azure DevOps [1] repo.

[1] https://dev.azure.com/marioloriedo/publicproject
[2] https://www.eclipse.org/che/docs/stable/administration-guide/configuring-oauth-2-for-microsoft-azure-devops-services/

### Describe the solution you'd like

Azure DevOps has its own OAuth page to implement in https://github.com/eclipse/che/blob/main/tests/e2e/pageobjects/git-providers/OauthPage.ts script

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by nallikaea**
### What does this PR do?
Resolve issue #22348 

### Screenshot/screencast of this PR

**_VIDEO:_** [Screencast from 2023-07-14 10-35-22.webm](https://github.com/eclipse/che/assets/72726250/850c63f2-ab58-407b-a010-a53014d20c9d)

Screenshot from 2023-07-14 10-38-43
Screenshot from 2023-07-13 14-01-53

###

 What issues does this PR fix or reference?

#22348 

### How to test this PR?

Set envs:
```
TS_SELENIUM_FACTORY_GIT_PROVIDER=azure
TS_SELENIUM_GIT_PROVIDER_OAUTH=true
TS_SELENIUM_FACTORY_GIT_REPO_URL=https://user@dev.azure.com/${path}
TS_SELENIUM_GIT_PROVIDER_USERNAME=${microsoft_acc_email}
TS_SELENIUM_GIT_PROVIDER_PASSWORD=${microsoft_acc_password}
```
Run defined spec file:
`USERSTORY=Factory && npm run test`

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix or new feature ](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix-or-new-feature)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by musienko-maxim — APPROVED** [review]

**Comment by nallikaea** [comment]
success pr check with EmptyWorkspace e2e test https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/Testing/job/e2e/job/basic/job/typescript-tests/9960/console

File: tests/e2e/package.json [hunk 1]
```diff
@@ -27,7 +27,7 @@
     "@types/shelljs": "^0.8.11",
     "axios": "^0.25.0",
     "chai": "^4.3.4",
-    "chromedriver": "^111.0.0",
+    "chromedriver": "^114.0.2",
```
Comment (ScrewTSW) [inline tests/e2e/package.json:30]: @nallikaea I don't seem to be able to find this change in the dockerfile as well. Please make sure the versions are consistent and the image builds successfully.

**Review by ScrewTSW — COMMENTED** [review]
Just one thing to check please

File: tests/e2e/package.json (hunk 1 above)
Comment (nallikaea) [reply tests/e2e/package.json:30]: In Dockerfile we still have `FROM selenium/standalone-chrome:107.0` is this line take any action? =)
https://github.com/eclipse/che/blob/3dd17c762c1f17ca85616418d2d886ac42d55b87/tests/e2e/build/dockerfiles/Dockerfile#L1

Comment (ScrewTSW) [reply tests/e2e/package.json:30]: yes, that should be updated if the package.json is being changed. oof. that's quite outdated then.

Comment (nallikaea) [reply tests/e2e/package.json:30]: updated

Comment (ScrewTSW) [reply tests/e2e/package.json:30]: Perfect, thank you!

**Review by ScrewTSW — APPROVED** [review]
LGTM

**Pull request merged**